### ISSUE TITLE
[ OJS ] Error in I5716_EmailTemplateAssignments from 3.3.0.8 to 3.4.0-3

### ISSUE THREAD

**Issue by rubrodapa** [pkp/pkp-lib#9518]
**Describe the bug**
Upgrading from 3.3.0.8 to 3.4.0-3 (also tested with 3.4.0-0) fails on APP\migration\upgrade\v3_4_0\InstallEmailTemplates.

ERROR: Upgrade failed: DB: Tried to install email template as an alternate to `DISCUSSION_NOTIFICATION_SUBMISSION`, but no default template exists with this key.

**To Reproduce**
Launch db check then upgrade

`php tools/upgrade.php check`

[Ok, no warn, no err]

`php tools/upgrade.php upgrade`

Results below

**What application are you using?**
ojs 3.3.0-8 attempt to upgrade 3.4.0-3

**Additional information**

Log before error

```
2023-11-16 12:58:15 [migration: APP\migration\upgrade\v3_4_0\I6782_MetricsIssue]
2023-11-16 12:58:18 [migration: APP\migration\upgrade\v3_4_0\I6782_MetricsSubmission]
2023-11-16 12:58:48 [migration: APP\migration\upgrade\v3_4_0\I6782_MetricsGeo]
2023-11-16 12:58:54 [migration: APP\migration\upgrade\v3_4_0\I6782_CleanOldMetrics]
2023-11-16 12:59:40 [migration: APP\migration\upgrade\v3_4_0\I6782_RemovePlugins]
2023-11-16 12:59:40 [migration: PKP\migration\upgrade\v3_4_0\I7286_BatchesMigration]
2023-11-16 12:59:40 [migration: APP\migration\upgrade\v3_4_0\I8151_ExtendSettingValues]
2023-11-16 12:59:42 [migration: PKP\migration\upgrade\v3_4_0\I8151_ExtendSettingValues]
2023-11-16 12:59:52 [migration: PKP\migration\upgrade\v3_4_0\I2890_EmailTemplatesVarcharLengthUpdate]
2023-11-16 12:59:53 [migration: PKP\migration\upgrade\v3_4_0\I2890_AddSetNullForOnDeleteToReviewRoundIdForeign]
2023-11-16 12:59:54 [migration: APP\migration\upgrade\v3_4_0\I7901_Duplicate_OAI_IDs]
2023-11-16 12:59:54 [migration: PKP\migration\upgrade\v3_4_0\I7366_UpdateUserAPIKeySettings]
2023-11-16 12:59:54 [migration: PKP\migration\upgrade\v3_4_0\I8093_UpdateUserGroupRelationTablesFK]
2023-11-16 12:59:55 [migration: APP\migration\upgrade\v3_4_0\I7725_DecisionConstantsUpdate]
2023-11-16 12:59:56 [migration: PKP\migration\upgrade\v3_4_0\I4789_AddReviewerRequestResentColumns]
2023-11-16 12:59:56 [migration: APP\migration\upgrade\v3_4_0\I7796_UpdateCrossrefSchema]
2023-11-16 12:59:56 [migration: PKP\migration\upgrade\v3_4_0\I7287_RemoveEmailTemplatesDefault]
2023-11-16 12:59:57 [migration: APP\migration\upgrade\v3_4_0\I5716_EmailTemplateAssignments]
2023-11-16 13:00:06 [revert migration: PKP\migration\upgrade\v3_4_0\I7287_RemoveEmailTemplatesDefault]
2023-11-16 13:00:18 [revert migration: APP\migration\upgrade\v3_4_0\I7796_UpdateCrossrefSchema]
2023-11-16 13:00:18 [downgrade for "APP\migration\upgrade\v3_4_0\I7796_UpdateCrossrefSchema" unsupported: Downgrade not supported]
ERROR: Upgrade failed: DB: Tried to install email template as an alternate to `DISCUSSION_NOTIFICATION_SUBMISSION`, but no default template exists with this key.
```

At the forum there are other users that also has the same problem but there is no answer: https://forum.pkp.sfu.ca/t/upgrade-failed-no-default-template-exists-with-this-key/79706

Any ideas how to solve this so we can upgrade? Thanks!

**Comment by asmecher** [comment]
@Vitaliy-1, could you have a look at this? Thanks!

**Comment by rubrodapa** [comment]
Thanks for your time @Vitaliy-1 . If I check at the OJS settings I can find an email templated called `NOTIFICATION_CENTER_DEFAULT`, but seems to have no data:

![image](https://github.com/pkp/pkp-lib/assets/3008433/7ed295fc-f24a-47be-b8c3-605d03d01130)

At the database, I can't find any template with email_key like `NOTIFICATION_CENTER_DEFAULT` neither at the `email_templates_default_data` nor the `email_templates_default`. Should I add it with some content?

**Comment by Vitaliy-1** [comment]
@rubrodapa, yes, it looks that the template data is missing. Do you have access to the command line? Before upgrading, it can be installed by running `php lib/pkp/tools/installEmailTemplate.php NOTIFICATION_CENTER_DEFAULT` from OJS root directory.

**Comment by Vitaliy-1** [comment]
PRs 
main
pkp-lib: https://github.com/pkp/pkp-lib/pull/9531
tests OJS: https://github.com/pkp/ojs/pull/4107

stable-3_4_0
pkp-lib: https://github.com/pkp/pkp-lib/pull/9532
tests OJS: https://github.com/pkp/ojs/pull/4108

**Comment by Vitaliy-1** [comment]
I ended up with the solution to directly install the default template in situations when it's missing and an alternate template depends on it. 

The original template `NOTIFICATION_CENTER_DEFAULT` is removed from the XML and replaced by one of the `DISCUSSION_NOTIFICATION_` templates as a part of 3.4 migration. The error occurs when `NOTIFICATION...` template is missing, then `DISCUSSION_NOTIFICATION_...` not being created by the migration code and the installation of other templates, which depend on them, triggers an error. 

@jonasraoni, can you take a look at the PRs?

**Comment by rubrodapa** [comment]
Hi. Sorry for the late response. I can confirm that installing the `NOTIFICATION_CENTER_DEFAULT ` makes the upgrade to work. Thanks for the help!

**Comment by jonasraoni** [comment]
@Vitaliy-1 I think these PRs are almost ready to merge, I just left two comments :)

**Comment by asmecher** [comment]
@Vitaliy-1 and @jonasraoni, I took the liberty of rushing the PRs into `main` (testing completed successfully) and `stable-3_4_0` (the first 3 finished and the rest looked fine) in order to get them included in 3.4.0-5.

**Comment by asmecher** [comment]
With all merged, please let me know if there's a reason I shouldn't be closing this!
